src/ExtMsg.actor.cpp

```diff
@@ -514,8 +514,15 @@ ACTOR static Future<Void> doRun(Reference<ExtMsgQuery> query, Reference<ExtConne
 		}
 	}
 
-	DocumentLayer::metricReporter->captureTime(DocLayerConstants::MT_TIME_QUERY_LATENCY_US,
-	                                           (timer_int() - startTime) / 1000);
+	uint64_t queryLatencySeconds = (timer_int() - startTime) / 1000;
```

Comment:
`time_init()` gives nanoseconds. That's why the metric is in microseconds, `US`, after division. This code fires slow query logging for every request taking more than 10 millies.